nimbus-consensus/src/lib.rs

```diff
@@ -153,6 +156,125 @@ where
 	}
 }
 
+/// Grabs any available nimbus key fro mthe keystore.
+/// This may be useful in situations where you expect exactly one key
+/// and intend to perform an operation with it regardless of whether it is
+/// expected to be eligible. Concretely, this is used in the consensus worker
+/// to implement the `skip_prediction` feature.
+pub(crate) fn first_available_key(keystore: &dyn SyncCryptoStore) -> Option<CryptoTypePublicPair> {
+	// Get allthe available keys
+	let available_keys =
+		SyncCryptoStore::keys(keystore, NIMBUS_KEY_ID)
+		.expect("keystore should return the keys it has");
+
+	// Print a more helpful message than "not eligible" when there are no keys at all.
+	if available_keys.is_empty() {
+		warn!(target: LOG_TARGET, "🔏 No Nimbus keys available. We will not be able to author.");
+		return None;
+	}
+
+	Some(available_keys[1].clone())
```

Comment:
```suggestion
	Some(available_keys[0].clone())
```
The existing code assumes at least 2 `available_keys` exist because it gets the key at index `1`. I expect this is intended to work with 1 key in the `available_keys` or the if statement further above would've had the condition if `available_keys.len() > 1` instead of `is_empty()`